lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1037,6 +1037,84 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     return DITy;
   }
 
+  /// Creates debug info for a generic struct with archetypes (eg.:
+  /// Pair<τ_0_0, τ_0_1>). For types with unsubstituted generic type paramaters,
+  /// debug info generation doesn't attempt to emit the size and aligment of
+  /// the type, as those are all dependent on substituting the type parameters
+  /// in. It also doesn't attempt to emit the offset of the members for the same
+  /// reason.
```

Comment:
This isn't strictly true if some of those type parameters are class-constrained, eg

```
protocol P {
  associatedtype A: AnyObject
}

struct Foo<T: P> {
  var x: T.A
}
```

Reply:
Thanks, I'll update the comment to be more accurate.